include/MySQL_HostGroups_Manager.h

```diff
@@ -390,6 +390,89 @@ class MySQL_HostGroups_Manager {
 #else
 	rwlock_t rwlock;
 #endif
+
+	enum HGM_TABLES {
+		MYSQL_SERVERS = 0,
+		MYSQL_REPLICATION_HOSTGROUPS,
+		MYSQL_GROUP_REPLICATION_HOSTGROUPS,
+		MYSQL_GALERA_HOSTGROUPS,
+		MYSQL_AWS_AURORA_HOSTGROUPS,
+		MYSQL_HOSTGROUP_ATTRIBUTES,
+
+		__HGM_TABLES_SIZE
+	};
+
+	std::array<uint64_t, __HGM_TABLES_SIZE> table_resultset_checksum;
+
+	class HostGroup_Server_Mapping {
```

Comment:
I don't see much benefit on of this structure being a class, since most of its members ends having a pair of trivial `getter/setter`  methods, this is usually an indication that there is no requirement for encapsulation, or that the interface exposed doesn't cover all the requirements for the type.

I think it also doesn't need to be a `nested` class of `MySQL_HostGroups_Manager`, since we are not in the need of hiding any implementation details of the upper class, I would understand if this was a library header, but it's not the case, please let me know if there is other motivations I'm missing.

Overall I think it could be simplified as an struct defined `MySQL_HostGroups_Manager`, with three functions to perform the special operations of `copy_servers_type`, `remove_server_type`, `clear_servers_type`. I see also okay the class encapsulation for keeping this relevant methods.

Reply:
In my opinion, it appears to be more suited towards a class. While it does have simple getters and setters, there are also methods such as copy_if_not_exists, remove, and clear, which involve unique record logic and updates to the mapping container. Additionally, the mapping container should be private to ensure proper encapsulation and prevent direct access without const. One key aspect of this code is that it includes a constructor.
The reason for nesting this class within MySQL_HostGroups_Manager is because it is currently only being used by that class. However, this could change in the future, as the removing dependency on hgm->mysql_servers table progresses.

Reply:
Closing as already discussed.